### DISCUSSION TITLE
Is Brython is planned to use Fetch API ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2425] [Q&A] [answered]
Hi,

According to https://developer.mozilla.org/en-US/docs/Glossary/XMLHttpRequest

> `Fetch API`  is the modern replacement for XMLHttpRequest.

ok, right... There is no explanation why..

I beleive the Brython `module browser.ajax` use [XHR](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) and it work quite well.

But would there be an advantage to use the `Fetch API` instead of XHR ? and is it planned to be implemented in Brython ?

Thanks :)

**Comment by PierreQuentel** [comment, marked as answer]
You can use `fetch()` like any other function in the global Javascript namespace:

```python
from browser import window, aio

async def fetch(url):
    resp = await window.fetch(url)
    print(await resp.text())

aio.run(fetch('test.html'))
```

**Reply by BobSquarePants** [reply]
Thank you @PierreQuentel ,

I will give it a try.. but will `fetch API` will perfome better/faster that XHR ? because I can't found any documentation about it.

Thanks.

**Comment by PierreQuentel** [comment]
It is documented like for other web technologies on the [MDN site](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).

I don't think it was designed for performance, but (quoting the MDN site) as a "a more powerful and flexible replacement for [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest).".
